ros/angel_system_nodes/angel_system_nodes/mm_activity_detector.py

```diff
@@ -0,0 +1,171 @@
+import json
+import time
+from threading import Lock
+from typing import Dict
+import pdb
+
+from cv_bridge import CvBridge
+import cv2
+import numpy as np
+import rclpy
+from rclpy.node import Node
+from rclpy.qos import qos_profile_sensor_data
+from smqtk_core.configuration import from_config_dict
+import message_filters as mf
+
+from sensor_msgs.msg import Image
+from angel_msgs.msg import ActivityDetection, HandJointPosesUpdate
+
+from angel_system.interfaces.mm_detect_activities import MMDetectActivities
+
+
+BRIDGE = CvBridge()
+    
+
+class MMActivityDetector(Node):
+
+    def __init__(self):
+        super().__init__(self.__class__.__name__)
+
+        self._image_topic = self.declare_parameter("image_topic", "/debug/PVFrames").get_parameter_value().string_value
+        self._hand_topic = self.declare_parameter("hand_pose_topic", "/debug/HandJointPoseData").get_parameter_value().string_value
+        self._use_cuda = self.declare_parameter("use_cuda", True).get_parameter_value().bool_value
+        self._det_topic = self.declare_parameter("det_topic", "ActivityDetections").get_parameter_value().string_value
+        self._frames_per_det = self.declare_parameter("frames_per_det", 32.0).get_parameter_value().double_value
+        self._detector_config = self.declare_parameter("detector_config", "default_activity_det_config.json").get_parameter_value().string_value
+
+        log = self.get_logger()
+        log.info(f"Image topic: {self._image_topic}")
+        log.info(f"Hand topic: {self._hand_topic}")
+        log.info(f"Use cuda? {self._use_cuda}")
+        log.info(f"Frames per detection: {self._frames_per_det}")
+        log.info(f"Detector config: {self._detector_config}")
+
+        self.subscription_list = []
+        # Image subscription
+        self.subscription_list.append(mf.Subscriber(self, Image, self._image_topic))
+        # Hand pose subscription
+        self.subscription_list.append(mf.Subscriber(self, HandJointPosesUpdate, self._hand_topic))
+        self.time_sync = mf.TimeSynchronizer(
```

Comment:
If we want to use `ApproximateTimeSynchronizer` then we will need to set a `slop` parameter that is the delay margin. Not sure what the correct margin is and this may also need extra checking like - single right/left hand within the slop interval and no more than 2 hands in the slop interval etc.